packages/core/src/lib/core/utils.ts

```diff
@@ -121,48 +121,35 @@ export async function promiseAllRejectWithAllErrors<T extends unknown[]>(
 
 export function getNamesFromList(
   listKey: string,
-  { graphql /*plural, label, singular, path */ }: KeystoneConfig['lists'][string]
+  { graphql, ui }: KeystoneConfig['lists'][string]
 ) {
-  const _label = /*label ||*/ keyToLabel(listKey);
-  const _singular = /*singular ||*/ pluralize.singular(_label);
-  const _plural = /*plural ||*/ pluralize.plural(_label);
+  const humanizedListKey = humanize(listKey);
+  const computedPlural = pluralize.plural(humanizedListKey);
 
-  if (_plural === _label) {
+  const path = ui?.path || labelToPath(computedPlural);
+
+  if (ui?.path !== undefined && !/^[a-z-_][a-z0-9-_]*$/.test(ui.path)) {
     throw new Error(
-      `Unable to use ${_label} as a List name - it has an ambiguous plural (${_plural}). Please choose another name for your list.`
+      `ui.path for ${listKey} is ${ui.path} but it must only contain lowercase letters, numbers, dashes, and underscores and not start with a number`
     );
   }
 
   const adminUILabels = {
-    // Fall back to the plural for the label if none was provided, not the autogenerated default from key
-    label: /*label ||*/ _plural,
-    singular: _singular,
-    plural: _plural,
-    path: /*path || */ labelToPath(_plural),
+    label: ui?.label || computedPlural,
+    singular: ui?.singular || pluralize.singular(humanizedListKey),
+    plural: ui?.plural || computedPlural,
+    path,
   };
 
-  const pluralGraphQLName = graphql?.plural || labelToClass(_plural);
+  const pluralGraphQLName = graphql?.plural || labelToClass(computedPlural);
   if (pluralGraphQLName === listKey) {
     throw new Error(
-      `The list key and the plural must be different but the list key ${listKey} is the same as the ${listKey} plural GraphQL name`
+      `The list key and the plural name used in GraphQL must be different but the list key ${listKey} is the same as the plural GraphQL name, please specify graphql.plural`
     );
   }
-  return {
-    pluralGraphQLName,
-    adminUILabels,
-  };
+  return { pluralGraphQLName, adminUILabels };
 }
 
-const keyToLabel = (str: string) => {
-  let label = humanize(str);
-
-  // Retain the leading underscore for auxiliary lists
-  if (str[0] === '_') {
-    label = `_${label}`;
-  }
```

Comment:
This was about a special kind of list that was magically created for certain things in Keystone 5 which does not exist in Keystone 6 so it's no longer relevant